omod/src/main/webapp/pages/labtrackingAddOrder.gsp

```diff
@@ -76,33 +77,29 @@ ${ ui.includeFragment("coreapps", "patientHeader", [ patient: patient ]) }
             <div class="form-group row">
                <label class="col-sm-2 col-form-label">${ui.message("labtrackingapp.prelabdiagnosislabel")}</label>
                <div class="col-sm-9">
-                  <input type="text" ng-model="order.preLabDiagnosis"  class="form-control" ng-required="false"
+                  <input ng-if="!order.uuid" type="text" ng-model="order.preLabDiagnosis"  class="form-control" ng-required="false"
                      uib-typeahead="item as item.label for item in alldiagnoses | filter:{label:${'$viewValue'}} | limitTo:8" />
+                  <p ng-if="order.uuid" class="form-control-static" > {{ formatDiagnosis(order.preLabDiagnosis) }}</p>
                </div>
             </div>
             <div class="form-group row">
                <label for="procedure" class="col-sm-2 col-form-label">${ui.message("labtrackingapp.proceduresitelabel")}</label>
                <div class="col-sm-4">
-                  <div class="panel panel-default">
-                     <div class="panel-heading">${ui.message("labtrackingapp.proceduresAvailable")}</div>
-                     <div class="panel-body">
-                        <select id="procedure" class="form-control" multiple
-                           ng-options="item as item.label for item in procedures | orderBy:'label' track by item.value "
-                           ng-model="selectedProcedures" ></select>
-                     </div>
-                  </div>
+                  <input  type="text" ng-model="selectedProcedures"  class="form-control" ng-required="false"
+                          uib-typeahead="item as item.label for item in procedures | filter:${'$viewValue'} | limitTo:8"
+                          typeahead-on-select="onSelectProcedure(${'$item'}, ${'model'}, ${'label'})"/>
                </div>
                <div class="col-sm-1">
                   <p>&nbsp;</p>
-                  <button class="btn button-wrapper" type="button" ng-disabled="order.procedures.length > 2" ng-click="addProcedure()"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span></button>
-                  <button class="btn button-wrapper" type="button" ng-disabled="order.procedures.length == 0" ng-click="removeProcedure()"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span></button>
+                  <button class="btn button-wrapper" type="button" ng-disabled="order.proceduresForSpecimen.length > 2" ng-click="addProcedure()"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span></button>
```

Comment:
Tangential, I noticed during testing that it didn't seem to allow more than 3 procedures to be ordered and I wondering if that was deliberate or not... looks like it was, so this must have been a requirement back in the day, not sure if it's worth asking the current team if they still want to limit to 3, or should we remove this restriction?

Reply:
Sure, this is easy to do. I will follow up with them in a email.